assets/js/components/ReportError.test.js

```diff
@@ -237,6 +237,79 @@ describe( 'ReportError', () => {
 		);
 	} );
 
+	it( 'renders alternate error for non-authenticated users when ERROR_REASON_INSUFFICIENT_PERMISSIONS is provided as reason', () => {
+		const { container } = render(
+			<ReportError
+				moduleSlug={ moduleName }
+				error={ {
+					code: 'test-error-code',
+					message: 'Test error message',
+					data: {
+						reason: ERROR_REASON_INSUFFICIENT_PERMISSIONS,
+					},
+				} }
+			/>,
+			{
+				registry,
+				viewContext: VIEW_CONTEXT_MAIN_DASHBOARD_VIEW_ONLY,
+			}
+		);
+
+		expect( container.querySelector( 'h3' ).textContent ).toEqual(
+			'Access lost to Test Module'
+		);
+	} );
+
+	it( 'should not render the `Request access` button even if it exists for a non-authenticated user', () => {
```

Comment:
This test description is a little confusing. If I have understood the test correctly, I think it would be clearer to say something like this:

```suggestion
	it( 'should not render the `Request access` button for an insufficient permission error when the user is not authenticated', () => {
```

Does that look right?

Reply:
Makes much more sense. Updated, thank you @techanvil!